mitmproxy/proxy/layers/http/_http2.py

```diff
@@ -407,7 +408,21 @@ def _handle_event(self, event: Event) -> CommandGenerator[None]:
                 yield from self._handle_event(event)
 
     def _handle_event2(self, event: Event) -> CommandGenerator[None]:
-        if isinstance(event, RequestHeaders):
+        if isinstance(event, Wakeup):
+            remaining = self.context.options.http2_ping_threshold - time.time() + self.last_activity
+            if remaining <= 0.5:
+                remaining = self.context.options.http2_ping_threshold
+                self.h2_conn.ping(b"0" * 8)
```

Comment:
I was wondering why you'd add 8 seemingly random bytes here. The RFC had the answer. :)
```suggestion
                # PING frames MUST contain 8 octets of opaque data in the payload. 
                # A sender can include any value it chooses and use those octets in any fashion.
                self.h2_conn.ping(b"0" * 8)
```